pkg/client/fake/client.go

```diff
@@ -78,7 +89,16 @@ func (c *fakeClient) Get(ctx context.Context, key client.ObjectKey, obj runtime.
 }
 
 func (c *fakeClient) List(ctx context.Context, opts *client.ListOptions, list runtime.Object) error {
-	gvk := opts.Raw.TypeMeta.GroupVersionKind()
+	gvk, err := getGVKFromList(list, c.scheme)
+	if err != nil {
+		// The old fake client required GVK info in Raw.TypeMeta, so check there
+		// before giving up
+		if opts.Raw.TypeMeta.APIVersion == "" || opts.Raw.TypeMeta.Kind == "" {
```

Comment:
Check if `opts.Raw == nil`, as that's what was causing the panic previously.

Reply:
Good spot @estroz, I accepted your suggestion.